pkg/apis/v1beta1/ec2nodeclass_hash_test.go

```diff
@@ -66,7 +67,21 @@ var _ = Describe("Hash", func() {
 		Entry("UserData Drift", "588756456110800812", v1beta1.EC2NodeClass{Spec: v1beta1.EC2NodeClassSpec{UserData: aws.String("userdata-test-2")}}),
 		Entry("Tags Drift", "2471764681523766508", v1beta1.EC2NodeClass{Spec: v1beta1.EC2NodeClassSpec{Tags: map[string]string{"keyTag-test-3": "valueTag-test-3"}}}),
 		Entry("MetadataOptions Drift", "11030161632375731908", v1beta1.EC2NodeClass{Spec: v1beta1.EC2NodeClassSpec{MetadataOptions: &v1beta1.MetadataOptions{HTTPEndpoint: aws.String("test-metadata-2")}}}),
-		Entry("BlockDeviceMappings Drift", "436753305915039702", v1beta1.EC2NodeClass{Spec: v1beta1.EC2NodeClassSpec{BlockDeviceMappings: []*v1beta1.BlockDeviceMapping{{DeviceName: aws.String("map-device-test-3")}}}}),
+		Entry("BlockDeviceMappings Drift", "2884646775270747279", v1beta1.EC2NodeClass{Spec: v1beta1.EC2NodeClassSpec{BlockDeviceMappings: []*v1beta1.BlockDeviceMapping{
```

Comment:
I think we should expanded this out make sure that making any changes to every field will cause the cache to change. I don't think this test would have helped us catch the issue we are trying to fix here.

Reply:
> I don't think this test would have helped us catch the issue we are trying to fix here

Test would have helped catch this specific issue, but I agree that we can expand this out to add additional testing at the field-level.

Reply:
Updated with the new tests.